docs/flatcar-linux/packet.md

```diff
@@ -328,5 +328,6 @@ Reference the DNS zone id with `"${aws_route53_zone.zone-for-clusters.zone_id}"`
 | cluster_domain_suffix | FQDN suffix for Kubernetes services answered by coredns. | "cluster.local" | "k8s.example.com" |
 | service_cidr | CIDR IPv4 range to assign to Kubernetes services | "10.3.0.0/16" | "10.3.0.0/24" |
 | setup_raid | Flag to create a RAID 0 from extra disks on a Packet node | false | true |
-| reservation_ids | Specify Packet hardware_reservation_id for instances, see: https://support.packet.com/kb/articles/reserved-hardware. A map where the key format is 'worker-${index}' and the associated value is the reservation id string. Nodes not present in the map will use no reservation id. | {} | { worker-0 = "55555f20-a1fb-55bd-1e11-11af11d11111" } |
 | taints | Comma separated list of custom taints for all workers in the worker pool | "" | "clusterType=staging:NoSchedule,nodeType=storage:NoSchedule" |
+| reservation_ids | Specify Packet hardware_reservation_id for instances, see: https://support.packet.com/kb/articles/reserved-hardware. A map where the key format is 'worker-${index}' and the associated value is the reservation id string. Nodes not present in the map will use no reservation id. | {} | { worker-0 = "55555f20-a1fb-55bd-1e11-11af11d11111" } |
+| reservation_ids_default | Value used for nodes not listed in the reservation_ids map. Note that using the empty string means using no hardware reservation and 'next-available' will choose any reservation that matches the instance type and facility this pool is running.| "" | "next-available"|
```

Comment:
Looking at https://www.terraform.io/docs/providers/packet/r/device.html#hardware_reservation_id, can't we just let the user know they can use `next-available` as a value in `reservation_ids`? Why do we need a separate variable?

Reply:
I suggest the following description instead of the existing one:

>Specify a default reservation ID for nodes not listed in the `reservation_ids` map. An empty string means "use no hardware reservation". `next-available` will choose any reservation that matches the worker pool's device type and facility.

Reply:
The other variable is a map, so they specify each node and a reservation id. Next-available can be used there, but if you want all the nodes in the worker pool to use "next-available" you need to list all the nodes. For example, if your worker pool has 16 nodes you need to:

```
reservation_ids = {
  worker-0 = "next-available"
  worker-1 = "next-available"
  worker-2 = "next-available"
 ...
}
```

And don't forget to set it when you add a new node.

With this new variable, if you want that setup, you can just do:

````
 reservation_ids_default = "next-available"
```

And obtain the same.

And if you, for some reason (rack placement inside packet for an instance or something) need to use a particular reservation ID for one node, you can still specify it with the map. For example, let's say you now need to create node 17 with "asd" as a reservation ID. You can:

```
 reservation_ids_default = "next-available"

 reservation_ids = {
 worker-16 = "asd"
 }

```

Is it more clear now?

Reply:
Yeah, I understand the mechanics, just wasn't sure about the use case.

Since `reservation_ids_default` has only two values (empty string or `next_available`), why not make this a boolean? I get the feeling it will simplify the whole thing.

I suggest renaming `reservation_ids_default` to `auto_assign_reservation_ids` and explain that it causes an automatic selection of reservation IDs while still allowing to explicitly specify a reservation ID to a device.

Reply:
> I suggest the following description instead of the existing one:
> 
> > Specify a default reservation ID for nodes not listed in the `reservation_ids` map. An empty string means "use no hardware reservation". `next-available` will choose any reservation that matches the worker pool's device type and facility.

Fixed this

Reply:
Oh, seems like a better abstraction for the Packet API, thanks!

I'm not sure we want this, though, because Packet may add other special value (as "next-available" is). If that happens, you can only specify one value in the "hardware_reservation_id" and this bool might become more messy and messier to deal with as that happens.

For example, if they add another value (whatever that is, let's say "next-available-with-closer-expiration-date") we might be creating a complicated path for us, because we will need to check if  `auto_assign_reservation_ids` is enabled and then choose the method. That is not such a big deal, but if they add more special values, we might fall into a trap.

I've mixed feeling with this, because although I really like the abstraction, the Packet API is something that is out of our control and not sure this abstractios makes sense or it will create more problems for us in the future.

What do you think? Others? @invidian ? :)

Reply:
> I suggest the following description instead of the existing one:
> 
> > Specify a default reservation ID for nodes not listed in the `reservation_ids` map. An empty string means "use no hardware reservation". `next-available` will choose any reservation that matches the worker pool's device type and facility.

Fixed this :)

Reply:
>I'm not sure we want this, though, because Packet may add other special value

You would have to make changes to the Terraform module anyway if Packet changes something. At the very least you would have to modify the docs to let the users know about the up-to-date possible values. IMO it's better to make things nice and clear with the information we have *now*, as long as we don't prevent ourselves from adapting in the future.

Reply:
@rata's point about future changes on the Packet side seems reasonable for me. If they add let's say rack reservation IDs or something. Changing string to boolean here seems like unnecessary optimization in this case. And if user puts bad value there, worst case Packet API complains, so it's not that bad. Also if something change in the future, the only thing we need to change is documentation. So I would leave it as it is.

Reply:
IMO the premature optimization is actually to do things now because Packet *may* change something in the future :-) But we can leave it as is, no strong opinions here as long as the docs are clear on how to use this.

Reply:
Great, then I'll leave this as it is. Thanks! :)